File: Models/Prayer.cs
using System.ComponentModel.DataAnnotations;

namespace MeTenTenMaui.Models
{
    public class Prayer
    {
        public int Id { get; set; }
        
        [Required(ErrorMessage = "제목을 입력해주세요.")]
        public string Title { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "내용을 입력해주세요.")]
        public string Content { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "카테고리를 선택해주세요.")]
        public string Category { get; set; } = string.Empty;
        
        public string? Tags { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public bool IsFavorite { get; set; }
        public int ViewCount { get; set; }
    }
}

File: webapp/src/types/prayer.ts
/**
 * 기도문 관련 타입 정의
 */

import { BaseEntity } from './common';

export interface Prayer extends BaseEntity {
  id: string;
  title: string;
  content: string;
  category: string;
  tags?: string[];
  isFavorite: boolean;
  viewCount: number;
}

export interface PrayerCategory {
  id: string;
  name: string;
  description?: string;
  prayers: Prayer[];
}

